txscript/script.go

```diff
@@ -278,6 +278,107 @@ func parseScriptTemplate(script []byte, opcodes *[256]opcode) ([]parsedOpcode, e
 	return retScript, nil
 }
 
+// checkScriptTemplateParseable is the same as parseScriptTemplate but does not
+// return the list of opcodes up until the point of failure so that this can be
+// used in functions which do not necessarily have a need for the failed list of
+// opcodes, such as IsUnspendable.
+// Not returning the full opcode list up until failure also has the benefit of
+// reducing GC pressure, as the list would get immediately thrown away.
+func checkScriptTemplateParseable(script []byte, opcodes *[256]opcode) (uint, byte, error) {
+	// A script of length zero is an unspendable script but it is parseable.
+	if len(script) == 0 {
+		return 0, 0x00, nil
+	}
+
+	var firstOpcode byte = OP_0
+	var numParsedInstr uint = 0
+	for i := 0; i < len(script); {
+		instr := script[i]
+		op := &opcodes[instr]
+		pop := parsedOpcode{opcode: op}
+
+		// Parse data out of instruction.
```

Comment:
The entire `switch` block appears to be duplicated in `parseScriptTemplate`. Do you think it's worth doing a refactoring?